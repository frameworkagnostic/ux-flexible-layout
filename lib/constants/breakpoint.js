'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var BREAKPOINT_KEYS = exports.BREAKPOINT_KEYS = ['xs', 'sm', 'md', 'lg', 'xl'];

var BREAKPOINTS = exports.BREAKPOINTS = {
  xs: 0,
  sm: 600,
  md: 960,
  lg: 1280,
  xl: 1920
};
//# sourceMappingURL=breakpoint.js.map