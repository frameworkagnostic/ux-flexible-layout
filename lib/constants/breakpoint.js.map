{"version":3,"sources":["../../src/constants/breakpoint.js"],"names":["BREAKPOINT_KEYS","BREAKPOINTS","xs","sm","md","lg","xl"],"mappings":";;;;;AACO,IAAMA,4CAAkB,CAC7B,IAD6B,EAE7B,IAF6B,EAG7B,IAH6B,EAI7B,IAJ6B,EAK7B,IAL6B,CAAxB;;AAQA,IAAMC,oCAAc;AACzBC,MAAI,CADqB;AAEzBC,MAAI,GAFqB;AAGzBC,MAAI,GAHqB;AAIzBC,MAAI,IAJqB;AAKzBC,MAAI;AALqB,CAApB","file":"breakpoint.js","sourcesContent":["\nexport const BREAKPOINT_KEYS = [\n  'xs',\n  'sm',\n  'md',\n  'lg',\n  'xl'\n];\n\nexport const BREAKPOINTS = {\n  xs: 0,\n  sm: 600,\n  md: 960,\n  lg: 1280,\n  xl: 1920,\n};\n"]}