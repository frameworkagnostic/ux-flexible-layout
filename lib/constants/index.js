'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GRID12_GUTTERS = exports.GRID12_SIZES = exports.STEP = exports.UNIT = exports.BREAKPOINTS = exports.BREAKPOINT_KEYS = undefined;

var _breakpoint = require('./breakpoint');

Object.defineProperty(exports, 'BREAKPOINT_KEYS', {
  enumerable: true,
  get: function get() {
    return _breakpoint.BREAKPOINT_KEYS;
  }
});
Object.defineProperty(exports, 'BREAKPOINTS', {
  enumerable: true,
  get: function get() {
    return _breakpoint.BREAKPOINTS;
  }
});

var _grid12Column = require('./grid12Column');

var grid12Column = _interopRequireWildcard(_grid12Column);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

var UNIT = exports.UNIT = 'px';
var STEP = exports.STEP = 5;

var GRID12_SIZES = exports.GRID12_SIZES = [].concat(_toConsumableArray(grid12Column.GRID_SIZES_DEFAULT), _toConsumableArray(grid12Column.GRID_SIZES));

var GRID12_GUTTERS = exports.GRID12_GUTTERS = grid12Column.GRID_GUTTERS;
//# sourceMappingURL=index.js.map