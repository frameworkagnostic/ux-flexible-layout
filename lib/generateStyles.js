'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateStyles = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _flexboxInlineStyles = require('./flexboxInlineStyles');

var _generateGutter = require('./generateGutter');

var _generateGridStyles = require('./generateGridStyles');

var _createBreakpoints2 = require('./createBreakpoints');

var _constants = require('./constants');

var generateStyles = exports.generateStyles = function generateStyles() {
  var grids = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _constants.GRID12_SIZES;
  var gutters = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _constants.GRID12_GUTTERS;

  var _createBreakpoints = (0, _createBreakpoints2.createBreakpoints)(),
      up = _createBreakpoints.up;

  return _extends({}, _flexboxInlineStyles.flexboxInlineStyles, gutters ? (0, _generateGutter.generateGutter)('xs', gutters) : {}, _createBreakpoints2.keys.reduce((0, _generateGridStyles.generateGridReducer)(grids, up), {}));
};
//# sourceMappingURL=generateStyles.js.map