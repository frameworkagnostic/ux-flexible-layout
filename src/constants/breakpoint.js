
export const BREAKPOINT_KEYS = [
  'xs',
  'sm',
  'md',
  'lg',
  'xl'
];

export const BREAKPOINTS = {
  xs: 0,
  sm: 600,
  md: 960,
  lg: 1280,
  xl: 1920,
};
