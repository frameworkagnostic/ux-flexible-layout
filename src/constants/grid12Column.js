
export const GRID_GUTTERS = [
  0,
  8,
  16,
  24,
  32,
  40
];

export const GRID_SIZES_DEFAULT = [
  'auto',
  true,
];

export const GRID_SIZES = [
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  11,
  12
];
