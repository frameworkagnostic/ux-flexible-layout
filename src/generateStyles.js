
import { flexboxInlineStyles } from './flexboxInlineStyles';
import { generateGutter } from './generateGutter';
import { generateGridReducer } from './generateGridStyles';
import { keys , createBreakpoints } from './createBreakpoints';
import { GRID12_SIZES, GRID12_GUTTERS } from './constants';

export const generateStyles = (grids = GRID12_SIZES, gutters = GRID12_GUTTERS) => {
  const { up } = createBreakpoints();
  const reducer = generateGridReducer(grids, up);

  return {
    ...flexboxInlineStyles,
    ...generateGutter('xs', gutters),
    ...keys.reduce(reducer, {}),
  };
}
