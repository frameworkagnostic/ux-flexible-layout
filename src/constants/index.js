
import * as grid12Column from './grid12Column';

export {
  BREAKPOINT_KEYS,
  BREAKPOINTS,
} from './breakpoint';

export const UNIT = 'px';
export const STEP = 5;

export const GRID12_SIZES = [
  ...grid12Column.GRID_SIZES_DEFAULT,
  ...grid12Column.GRID_SIZES,
];

export const GRID12_GUTTERS = grid12Column.GRID_GUTTERS;
