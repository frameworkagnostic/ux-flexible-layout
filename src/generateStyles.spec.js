
import { generateStyles } from './generateStyles';
import { GRID12_GUTTERS } from './constants';

describe('generate styles', () => {
  test('isFunction', () => expect(generateStyles).toEqual(expect.any(Function)));

  describe('default grid styles', () => {
    const styles = generateStyles();
    test('snapshot', () => expect(styles).toMatchSnapshot());
  });
});
