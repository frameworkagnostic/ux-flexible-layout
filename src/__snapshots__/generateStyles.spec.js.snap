// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate styles default grid styles snapshot 1`] = `
Object {
  "@media (min-width:1280px)": Object {
    "grid-lg-1": Object {
      "flexBasis": "8.33333%",
      "flexGrow": 0,
      "maxWidth": "8.33333%",
    },
    "grid-lg-10": Object {
      "flexBasis": "83.33333%",
      "flexGrow": 0,
      "maxWidth": "83.33333%",
    },
    "grid-lg-11": Object {
      "flexBasis": "91.66667%",
      "flexGrow": 0,
      "maxWidth": "91.66667%",
    },
    "grid-lg-12": Object {
      "flexBasis": "100%",
      "flexGrow": 0,
      "maxWidth": "100%",
    },
    "grid-lg-2": Object {
      "flexBasis": "16.66667%",
      "flexGrow": 0,
      "maxWidth": "16.66667%",
    },
    "grid-lg-3": Object {
      "flexBasis": "25%",
      "flexGrow": 0,
      "maxWidth": "25%",
    },
    "grid-lg-4": Object {
      "flexBasis": "33.33333%",
      "flexGrow": 0,
      "maxWidth": "33.33333%",
    },
    "grid-lg-5": Object {
      "flexBasis": "41.66667%",
      "flexGrow": 0,
      "maxWidth": "41.66667%",
    },
    "grid-lg-6": Object {
      "flexBasis": "50%",
      "flexGrow": 0,
      "maxWidth": "50%",
    },
    "grid-lg-7": Object {
      "flexBasis": "58.33333%",
      "flexGrow": 0,
      "maxWidth": "58.33333%",
    },
    "grid-lg-8": Object {
      "flexBasis": "66.66667%",
      "flexGrow": 0,
      "maxWidth": "66.66667%",
    },
    "grid-lg-9": Object {
      "flexBasis": "75%",
      "flexGrow": 0,
      "maxWidth": "75%",
    },
    "grid-lg-auto": Object {
      "flexBasis": "auto",
      "flexGrow": 0,
      "maxWidth": "none",
    },
    "grid-lg-true": Object {
      "flexBasis": 0,
      "flexGrow": 1,
      "maxWidth": "100%",
    },
  },
  "@media (min-width:1920px)": Object {
    "grid-xl-1": Object {
      "flexBasis": "8.33333%",
      "flexGrow": 0,
      "maxWidth": "8.33333%",
    },
    "grid-xl-10": Object {
      "flexBasis": "83.33333%",
      "flexGrow": 0,
      "maxWidth": "83.33333%",
    },
    "grid-xl-11": Object {
      "flexBasis": "91.66667%",
      "flexGrow": 0,
      "maxWidth": "91.66667%",
    },
    "grid-xl-12": Object {
      "flexBasis": "100%",
      "flexGrow": 0,
      "maxWidth": "100%",
    },
    "grid-xl-2": Object {
      "flexBasis": "16.66667%",
      "flexGrow": 0,
      "maxWidth": "16.66667%",
    },
    "grid-xl-3": Object {
      "flexBasis": "25%",
      "flexGrow": 0,
      "maxWidth": "25%",
    },
    "grid-xl-4": Object {
      "flexBasis": "33.33333%",
      "flexGrow": 0,
      "maxWidth": "33.33333%",
    },
    "grid-xl-5": Object {
      "flexBasis": "41.66667%",
      "flexGrow": 0,
      "maxWidth": "41.66667%",
    },
    "grid-xl-6": Object {
      "flexBasis": "50%",
      "flexGrow": 0,
      "maxWidth": "50%",
    },
    "grid-xl-7": Object {
      "flexBasis": "58.33333%",
      "flexGrow": 0,
      "maxWidth": "58.33333%",
    },
    "grid-xl-8": Object {
      "flexBasis": "66.66667%",
      "flexGrow": 0,
      "maxWidth": "66.66667%",
    },
    "grid-xl-9": Object {
      "flexBasis": "75%",
      "flexGrow": 0,
      "maxWidth": "75%",
    },
    "grid-xl-auto": Object {
      "flexBasis": "auto",
      "flexGrow": 0,
      "maxWidth": "none",
    },
    "grid-xl-true": Object {
      "flexBasis": 0,
      "flexGrow": 1,
      "maxWidth": "100%",
    },
  },
  "@media (min-width:600px)": Object {
    "grid-sm-1": Object {
      "flexBasis": "8.33333%",
      "flexGrow": 0,
      "maxWidth": "8.33333%",
    },
    "grid-sm-10": Object {
      "flexBasis": "83.33333%",
      "flexGrow": 0,
      "maxWidth": "83.33333%",
    },
    "grid-sm-11": Object {
      "flexBasis": "91.66667%",
      "flexGrow": 0,
      "maxWidth": "91.66667%",
    },
    "grid-sm-12": Object {
      "flexBasis": "100%",
      "flexGrow": 0,
      "maxWidth": "100%",
    },
    "grid-sm-2": Object {
      "flexBasis": "16.66667%",
      "flexGrow": 0,
      "maxWidth": "16.66667%",
    },
    "grid-sm-3": Object {
      "flexBasis": "25%",
      "flexGrow": 0,
      "maxWidth": "25%",
    },
    "grid-sm-4": Object {
      "flexBasis": "33.33333%",
      "flexGrow": 0,
      "maxWidth": "33.33333%",
    },
    "grid-sm-5": Object {
      "flexBasis": "41.66667%",
      "flexGrow": 0,
      "maxWidth": "41.66667%",
    },
    "grid-sm-6": Object {
      "flexBasis": "50%",
      "flexGrow": 0,
      "maxWidth": "50%",
    },
    "grid-sm-7": Object {
      "flexBasis": "58.33333%",
      "flexGrow": 0,
      "maxWidth": "58.33333%",
    },
    "grid-sm-8": Object {
      "flexBasis": "66.66667%",
      "flexGrow": 0,
      "maxWidth": "66.66667%",
    },
    "grid-sm-9": Object {
      "flexBasis": "75%",
      "flexGrow": 0,
      "maxWidth": "75%",
    },
    "grid-sm-auto": Object {
      "flexBasis": "auto",
      "flexGrow": 0,
      "maxWidth": "none",
    },
    "grid-sm-true": Object {
      "flexBasis": 0,
      "flexGrow": 1,
      "maxWidth": "100%",
    },
  },
  "@media (min-width:960px)": Object {
    "grid-md-1": Object {
      "flexBasis": "8.33333%",
      "flexGrow": 0,
      "maxWidth": "8.33333%",
    },
    "grid-md-10": Object {
      "flexBasis": "83.33333%",
      "flexGrow": 0,
      "maxWidth": "83.33333%",
    },
    "grid-md-11": Object {
      "flexBasis": "91.66667%",
      "flexGrow": 0,
      "maxWidth": "91.66667%",
    },
    "grid-md-12": Object {
      "flexBasis": "100%",
      "flexGrow": 0,
      "maxWidth": "100%",
    },
    "grid-md-2": Object {
      "flexBasis": "16.66667%",
      "flexGrow": 0,
      "maxWidth": "16.66667%",
    },
    "grid-md-3": Object {
      "flexBasis": "25%",
      "flexGrow": 0,
      "maxWidth": "25%",
    },
    "grid-md-4": Object {
      "flexBasis": "33.33333%",
      "flexGrow": 0,
      "maxWidth": "33.33333%",
    },
    "grid-md-5": Object {
      "flexBasis": "41.66667%",
      "flexGrow": 0,
      "maxWidth": "41.66667%",
    },
    "grid-md-6": Object {
      "flexBasis": "50%",
      "flexGrow": 0,
      "maxWidth": "50%",
    },
    "grid-md-7": Object {
      "flexBasis": "58.33333%",
      "flexGrow": 0,
      "maxWidth": "58.33333%",
    },
    "grid-md-8": Object {
      "flexBasis": "66.66667%",
      "flexGrow": 0,
      "maxWidth": "66.66667%",
    },
    "grid-md-9": Object {
      "flexBasis": "75%",
      "flexGrow": 0,
      "maxWidth": "75%",
    },
    "grid-md-auto": Object {
      "flexBasis": "auto",
      "flexGrow": 0,
      "maxWidth": "none",
    },
    "grid-md-true": Object {
      "flexBasis": 0,
      "flexGrow": 1,
      "maxWidth": "100%",
    },
  },
  "align-content-xs-center": Object {
    "alignContent": "center",
  },
  "align-content-xs-flex-end": Object {
    "alignContent": "flex-end",
  },
  "align-content-xs-flex-start": Object {
    "alignContent": "flex-start",
  },
  "align-content-xs-space-around": Object {
    "alignContent": "space-around",
  },
  "align-content-xs-space-between": Object {
    "alignContent": "space-between",
  },
  "align-items-xs-baseline": Object {
    "alignItems": "baseline",
  },
  "align-items-xs-center": Object {
    "alignItems": "center",
  },
  "align-items-xs-flex-end": Object {
    "alignItems": "flex-end",
  },
  "align-items-xs-flex-start": Object {
    "alignItems": "flex-start",
  },
  "container": Object {
    "boxSizing": "border-box",
    "display": "flex",
    "flexWrap": "wrap",
    "width": "100%",
  },
  "direction-xs-column": Object {
    "flexDirection": "column",
  },
  "direction-xs-column-reverse": Object {
    "flexDirection": "column-reverse",
  },
  "direction-xs-row-reverse": Object {
    "flexDirection": "row-reverse",
  },
  "grid-xs-1": Object {
    "flexBasis": "8.33333%",
    "flexGrow": 0,
    "maxWidth": "8.33333%",
  },
  "grid-xs-10": Object {
    "flexBasis": "83.33333%",
    "flexGrow": 0,
    "maxWidth": "83.33333%",
  },
  "grid-xs-11": Object {
    "flexBasis": "91.66667%",
    "flexGrow": 0,
    "maxWidth": "91.66667%",
  },
  "grid-xs-12": Object {
    "flexBasis": "100%",
    "flexGrow": 0,
    "maxWidth": "100%",
  },
  "grid-xs-2": Object {
    "flexBasis": "16.66667%",
    "flexGrow": 0,
    "maxWidth": "16.66667%",
  },
  "grid-xs-3": Object {
    "flexBasis": "25%",
    "flexGrow": 0,
    "maxWidth": "25%",
  },
  "grid-xs-4": Object {
    "flexBasis": "33.33333%",
    "flexGrow": 0,
    "maxWidth": "33.33333%",
  },
  "grid-xs-5": Object {
    "flexBasis": "41.66667%",
    "flexGrow": 0,
    "maxWidth": "41.66667%",
  },
  "grid-xs-6": Object {
    "flexBasis": "50%",
    "flexGrow": 0,
    "maxWidth": "50%",
  },
  "grid-xs-7": Object {
    "flexBasis": "58.33333%",
    "flexGrow": 0,
    "maxWidth": "58.33333%",
  },
  "grid-xs-8": Object {
    "flexBasis": "66.66667%",
    "flexGrow": 0,
    "maxWidth": "66.66667%",
  },
  "grid-xs-9": Object {
    "flexBasis": "75%",
    "flexGrow": 0,
    "maxWidth": "75%",
  },
  "grid-xs-auto": Object {
    "flexBasis": "auto",
    "flexGrow": 0,
    "maxWidth": "none",
  },
  "grid-xs-true": Object {
    "flexBasis": 0,
    "flexGrow": 1,
    "maxWidth": "100%",
  },
  "item": Object {
    "boxSizing": "border-box",
    "margin": "0",
  },
  "justify-xs-center": Object {
    "justifyContent": "center",
  },
  "justify-xs-flex-end": Object {
    "justifyContent": "flex-end",
  },
  "justify-xs-space-around": Object {
    "justifyContent": "space-around",
  },
  "justify-xs-space-between": Object {
    "justifyContent": "space-between",
  },
  "spacing-xs-16": Object {
    "& > $item": Object {
      "padding": 8,
    },
    "margin": -8,
    "width": "calc(100% + 16px)",
  },
  "spacing-xs-24": Object {
    "& > $item": Object {
      "padding": 12,
    },
    "margin": -12,
    "width": "calc(100% + 24px)",
  },
  "spacing-xs-32": Object {
    "& > $item": Object {
      "padding": 16,
    },
    "margin": -16,
    "width": "calc(100% + 32px)",
  },
  "spacing-xs-40": Object {
    "& > $item": Object {
      "padding": 20,
    },
    "margin": -20,
    "width": "calc(100% + 40px)",
  },
  "spacing-xs-8": Object {
    "& > $item": Object {
      "padding": 4,
    },
    "margin": -4,
    "width": "calc(100% + 8px)",
  },
  "wrap-xs-nowrap": Object {
    "flexWrap": "nowrap",
  },
  "wrap-xs-wrap-reverse": Object {
    "flexWrap": "wrap-reverse",
  },
  "zeroMinWidth": Object {
    "minWidth": 0,
  },
}
`;
